torchx/CMakeLists.txt

```diff
@@ -6,39 +6,91 @@ if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
   set(CMAKE_CUDA_ARCHITECTURES 75)
 endif()
 
+if(NOT DEFINED ENV{CMAKE_TOOLCHAIN_FILE})
+  set(CMAKE_TOOLCHAIN_FILE $ENV{CMAKE_TOOLCHAIN_FILE})
+endif()
+
+if(NOT DEFINED CMAKE_BUILD_TYPE OR "${CMAKE_BUILD_TYPE}" STREQUAL "")
+    set(CMAKE_BUILD_TYPE "Release")
+endif()
+
+if(DEFINED ENV{MIX_APP_PATH})
+    set(PRIV_DIR "$ENV{MIX_APP_PATH}/priv")
+else()
+    set(PRIV_DIR "${CMAKE_CURRENT_SOURCE_DIR}/priv")
+endif()
+set(CMAKE_INSTALL_PREFIX "${PRIV_DIR}")
+if(NOT IS_DIRECTORY "${PRIV_DIR}")
+  file(MAKE_DIRECTORY "${PRIV_DIR}")
+endif()
+
+if(DEFINED ERTS_INCLUDE_DIR AND NOT "${ERTS_INCLUDE_DIR}" STREQUAL "")
+    set(ERTS_INCLUDE_DIR "${ERTS_INCLUDE_DIR}")
+else()
+    set(ERTS_INCLUDE_DIR_ONE_LINER "erl -noshell -eval \"io:format('~ts/erts-~ts/include/', [code:root_dir(), erlang:system_info(version)]), halt().\"")
+    if(WIN32)
+        execute_process(COMMAND powershell -command "${ERTS_INCLUDE_DIR_ONE_LINER}" OUTPUT_VARIABLE ERTS_INCLUDE_DIR)
+    else()
+        execute_process(COMMAND bash -c "${ERTS_INCLUDE_DIR_ONE_LINER}" OUTPUT_VARIABLE ERTS_INCLUDE_DIR)
+    endif()
+    set(ERTS_INCLUDE_DIR "${ERTS_INCLUDE_DIR}")
+endif()
+include_directories(${ERTS_INCLUDE_DIR})
+
+set(C_SRC "${CMAKE_CURRENT_SOURCE_DIR}/c_src")
+set(Torch_DIR "$ENV{LIBTORCH_DIR}/share/cmake/Torch")
+
+message(STATUS "C_SRC: ${C_SRC}")
+message(STATUS "MIX_APP_PATH: $ENV{MIX_APP_PATH}")
+message(STATUS "PRIV_DIR: ${PRIV_DIR}")
+message(STATUS "ERTS_INCLUDE_DIR: ${ERTS_INCLUDE_DIR}")
+message(STATUS "LIBTORCH_DIR: $ENV{LIBTORCH_DIR}")
+message(STATUS "LIBTORCH_BASE: $ENV{LIBTORCH_BASE}")
+message(STATUS "MIX_BUILD_EMBEDDED $ENV{MIX_BUILD_EMBEDDED}")
+message(STATUS "LIBTORCH_LINK $ENV{LIBTORCH_LINK}")
+
+if((NOT DEFINED ENV{LIBTORCH_DIR}) OR (NOT IS_DIRECTORY $ENV{LIBTORCH_DIR}))
```

Comment:
We can also move this check to our `cmake` function? :)

Reply:
Will do!